File: src/Crates.Api/Models/Track.cs
using System;
using System.Collections.Generic;

namespace Crates.Api.Models
{
    public class Track
    {
        public Guid TrackId { get; private set; }
        public string Name { get; private set; }
        public List<Artist> Artists { get; private set; } = new();
        public List<Playlist> Playlists { get; private set; } = new();
        public string Spotify { get; private set; }
        public string AppleMusic { get; private set; }
        public Guid CoverArtDigitalAssetId { get; private set; }
    }
}

File: src/Crates.App/src/app/workspace/tracks/track.ts
import { Artist } from "../artists/artist";
import { Playlist } from "../playlists/playlist";

export type Track = {
    trackId:string;
    name: string;
    artists: Artist[];
    artistName: string;
    albumName: string;
    playlists: Playlist[];
    spotify: string;
    appleMusic: string;
    coverArtDigitalAssetId: string;
};
